Functional.Common.Extensions/ReplayableEnumerable.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+
+namespace Functional.Common.Extensions
```

Comment:
I took a closer look at the existing code, and I already have duplicate names even for public classes that contain extension methods in several places. Whether or not that was a good idea is definitely up for debate, but it seems to be fine to have the collisions. Fully qualified type names include the assembly name, so that must be why it's fine.

That being the case, I'd prefer this be changed back to just Functional for consistency with the other internal classes.